src/addons/vue-directives.ts

```diff
@@ -0,0 +1,188 @@
+import type { Addon, AddonVueDirectivesOptions, Import } from '../types'
+import { basename } from 'node:path'
+import process from 'node:process'
+import { resolve } from 'pathe'
+import { camelCase, kebabCase } from 'scule'
+import { stringifyImports } from '../utils'
+
+const contextRE = /resolveDirective as _resolveDirective/
+const contextText = `${contextRE.source}, `
+const directiveRE = /(?:var|const) (\w+) = _resolveDirective\("([\w.-]+)"\);?\s*/g
+
+export const VUE_DIRECTIVES_NAME = 'unimport:vue-directives'
+
+export function vueDirectivesAddon(
+  options: AddonVueDirectivesOptions = {},
+): Addon {
+  function isDirective(importEntry: Import) {
+    return importEntry.meta?.vueDirective === true
+      || (options.isDirective?.(normalizePath(process.cwd(), importEntry.from), importEntry) ?? false)
+  }
+
+  const self = {
+    name: VUE_DIRECTIVES_NAME,
+    async transform(s, id) {
+      if (!s.original.includes('_ctx.') || !s.original.match(contextRE))
+        return s
+
+      const matches = Array
+        .from(s.original.matchAll(directiveRE))
+        .sort((a, b) => b.index - a.index)
+
+      if (!matches.length)
+        return s
+
+      let targets: Import[] = []
+      for await (
+        const [
+          begin,
+          end,
+          importEntry,
+        ] of findDirectives(
+          isDirective,
+          matches,
+          this.getImports(),
+        )
+      ) {
+        // remove the directive declaration
+        s.overwrite(begin, end, '')
+        targets.push(importEntry)
+      }
+
+      if (!targets.length)
+        return s
+
+      // remove resolveDirective import
+      s.replace(contextText, '')
+
+      for (const addon of this.addons) {
+        if (addon === self)
+          continue
+
+        targets = await addon.injectImportsResolved?.call(this, targets, s, id) ?? targets
+      }
+
+      let injection = stringifyImports(targets)
+      for (const addon of this.addons) {
+        if (addon === self)
+          continue
+
+        injection = await addon.injectImportsStringified?.call(this, injection, targets, s, id) ?? injection
+      }
+
+      s.prepend(injection)
+
+      return s
+    },
+    async declaration(dts, options) {
+      const directivesMap = await this.getImports().then((imports) => {
+        return imports.filter(isDirective).reduce((acc, i) => {
+          if (i.type || i.dtsDisabled)
+            return acc
+
+          let name: string
+          if (i.name === 'default' && (i.as === 'default' || !i.as)) {
+            const file = basename(i.from)
+            const idx = file.indexOf('.')
+            name = camelCase(`v-${idx > -1 ? file.slice(0, idx) : file}`)
+          }
+          else {
+            name = camelCase(`v-${i.as ?? i.name}`)
```

Comment:
check if the named export also follows vue convention: https://vuejs.org/guide/reusability/custom-directives#introduction